javascript/core/src/verbs/merge.ts

```diff
@@ -49,7 +62,46 @@ export const mergeStep: ColumnTableStep<MergeArgs> = (
 		}
 	})
 
-	return input.derive({ [to]: func })
+	if (keepOriginalColumns) return tempTable.derive({ [to]: func })
+
+	return tempTable.derive({ [to]: func }).select(not(columns))
+}
+
+function unhotOperation(
+	input: ColumnTable,
+	columns: string[],
+	prefix: string,
+): ColumnTable {
+	const inputTable: ColumnTable = fromJSON(input.toJSON())
```

Comment:
You should be able to use `inputTable.reify()` here, which should perform better